### ISSUE TITLE
[v.Next,EKS] Velero Helm Chart

### ISSUE THREAD

**Issue by ph-One** [department-of-veterans-affairs/va.gov-team#85707]
## Description
Install the Velero helm chart.

## Resources 
- https://velero.io/
- https://github.com/department-of-veterans-affairs/vsp-platform-infrastructure/tree/main/terraform/modules/platform-distro

## Acceptance Criteria
- [x] Velero is up and running in Platform v.Next
- [x] Test backups via the `schedules.velero.io` CRD

**Comment by adhoc-nshipman** [comment]
I'll be shelving this ticket for the meantime. 

I'll need more time to determine the best way to implement Velero as currently the best approach seems to be done via the velero cli. If we want Velero installed and backing up via automation, we will need to leverage local exec to install the operator. 

This approach goes against how we normally onboard new applications to a cluster. For now, I will deprioritize this until I have a better scope of our disaster recovery needs and will switch off to autoscaling work. 

@kflow85 @ph-One

**Comment by adhoc-nshipman** [comment]
So I might be blocked on this issue, until I can find a work around or upgrade the chart: 

https://github.com/vmware-tanzu/velero/issues/8059

```
╷
│ Error: 2 errors occurred:
│       * BackupStorageLocation.velero.io "default" is invalid: [spec.objectStorage.bucket: Required value, spec.provider: Required value]
│       * VolumeSnapshotLocation.velero.io "default" is invalid: spec.provider: Required value
```

**Comment by adhoc-nshipman** [comment]
Current PR with progress (Do not Merge): https://github.com/department-of-veterans-affairs/vsp-platform-infrastructure/pull/691

**Comment by jwoodman5** [comment]
@kflow85 and/or @adhoc-nshipman I am working through sprint reporting. Can you please drop a comment about why this was not completed and what is happening with it (i.e. rolling over to next sprint, etc.)? Thanks.

**Comment by adhoc-nshipman** [comment]
1. Progress has been delayed and I had to create and push images to our private ECR due to rate limiting from Docker hub when pulling images down.

2. Still working through an issue (posted above) with the provider configurations failing when setting up aws as a backup storage location and the volume snapshot location. If I am unable to resolve the issue by Friday, I will put it on hold and clean up my PR to only deploy a version of Velero that does not use AWS as a provider.

**Comment by JoeTice** [comment]
Sprint 21 Update: This ticket is in progress but experienced a temporary delay this sprint due to unexpected resource constraints. Work will resume in the upcoming sprint, and we'll reassess the timeline during our next planning session. The ticket priority remains unchanged.

**Comment by JoeTice** [comment]
@adhoc-nshipman - Please provide a status update of this issue so we can plan next steps

**Comment by adhoc-nshipman** [comment]
Current updates: 

## Issues
- Ran into issues with local containers not being built and deployed with the proper architecture with led to the pod crashes in the live k8s cluster.
- Helm values for velero defined in platform distro are not being applied to the chart. Opted to using the live charts instead of rolling out are own for now.
- We are running into rate limiting for docker hub when using the official images for velero. Until we resolve the issue with building our own images (likely better through CI) we should add a docker hub token to each namespace to allow applications to pull from docker hub.

## Notes
- The fix for the image should be to create the docker hub auth token in the cluster and have velero pull the official image.

PR (WIP): https://github.com/department-of-veterans-affairs/vsp-platform-infrastructure/pull/691/files

**Comment by JoeTice** [comment]
**Sprint 22 Update: On Hold** – Work on this ticket is on hold due to resource transition. We anticipate reassignment during Sprint 23.

**Comment by stoiven** [comment]
update: this won't be done in this sprint, as there still needs to be discovery work involved with installing the helm chart. i've reviewed the PR, but am running into similar issues with what norman had had. this ticket is **nice-to-have** but not a blocker for other future work/projects

this should get pushed to the backlog

**Comment by JoeTice** [comment]
**Sprint 23 Update: Deprioritized** - Technical challenges with container architecture and Helm values require additional discovery. Reassessing approach and moving to backlog.

**Comment by stoiven** [comment]
```
next(steven-velero) > kubectl get schedules -n velero
NAME            STATUS    SCHEDULE    LASTBACKUP   AGE     PAUSED
velero-backup   Enabled   0 * * * *                6m24s   
```
